### PR TITLE
Fix issues in publisher related to AsyncAPI

### PR DESCRIPTION
**Pull request by lasanthaS**
This PR fixes issues in Publisher/DevPortal apps related to AsyncAPI.

Fixes https://github.com/wso2/product-apim/issues/10513
Fixes https://github.com/wso2/product-apim/issues/10585
Fixes https://github.com/wso2/product-apim/issues/10188
Fixes https://github.com/wso2/product-apim/issues/10128
Fixes https://github.com/wso2/product-apim/issues/10671
Fixes https://github.com/wso2/product-apim/issues/10465
Fixes https://github.com/wso2/product-apim/issues/10676

### CODE REVIEW

**Comment by malinthaprasan** [comment]
Please merge with latest master and update the PR

**Review by tmkasun — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 1]
```diff
@@ -1288,6 +1288,8 @@ public static GenericArtifact createAPIArtifactContent(GenericArtifact artifact,
             artifact.setAttribute(APIConstants.API_OVERVIEW_WEBSUB_SUBSCRIPTION_CONFIGURATION,
                     APIUtil.getWebsubSubscriptionConfigurationJsonFromDto(api.getWebsubSubscriptionConfiguration()));
 
+            artifact.setAttribute(APIConstants.API_OVERVIEW_WS_URI_MAPPING, APIUtil.getWsUriMappingJsonFromDto(api.getWsUriMapping()));
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:1291]: this needs to be happening at apimgt.persistence implementation and is there any rxt level change for this?.

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 1 above)
Comment (lasanthaS) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:1291]: RXT level change is already there in the master branch.

Comment (lasanthaS) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:1291]: And the CORS configurations are set here as well. So followed the same.

Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:1291]: This method is not used. Following is the usage of this method.

Screenshot 2021-04-02 at 5 31 32 PM

createAPI() method is not used when creating the API. check addAPI() method for this
updateWSDL methods are not used as well. that logic is in the updateAPI() method

Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:1291]: this content should be in addAPI() method in RegistryPersistenceImpl class

Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:1291]: please check the comment

Comment (lasanthaS) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:1291]: The relevant changes are already there in the RegistryPersistenceImpl class. Seems this is a redundant fix.

**Comment by malinthaprasan** [comment]
Workflows: https://github.com/lasanthaS/carbon-apimgt/actions/runs/711769175

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/pom.xml [hunk 2]
```diff
@@ -147,6 +147,10 @@
             <classifier>runtime</classifier>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.yaml</groupId>
+            <artifactId>snakeyaml</artifactId>
+        </dependency>
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/pom.xml:153]: Please add `<scope>provided</scope>`

Comment (malinthaprasan) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/pom.xml:153]: can add later

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**